src/store.js

```diff
@@ -26,24 +26,51 @@ export default new Vuex.Store({
         description: "Very nice cup"
       }
     ],
-    cart: [{
+    cart: {
       orderId: 1,
       status: 1,
-      products: [
-
-      ]
-    }],
+      count: 0,
+      products: []
+    },
     taxonomies: {
 
     }
   },
   mutations: {
     TOGGLE_MENU(state) {
-      state.menu.visible = !state.menu.visible;
+      state.menu.visible = !state.menu.visible
     },
     ADD_PRODUCT_TO_STORAGE(state, product) {
-      state.nextProductId++;
+      state.nextProductId++
       state.allProducts.push(product)
+    },
+    REMOVE_PRODUCT_FROM_CART(state, productId) {
+      state.cart.products.forEach((item, index) => {
+        if (item.productId == productId) {
+          state.cart.products.splice(index, 1)
+          state.cart.count--;
+        }
+      })
+    },
+    ADD_PRODUCT_TO_CART(state, {
+      productId,
+      quantity
+    }) {
+      let itemExist = false;
+      state.cart.products.forEach((item, index) => {
+        if (item.productId == productId) {
+          item.quantity += quantity
+          itemExist = true;
+        }
+      })
+
+      if (!itemExist) {
+        state.cart.products.push({
+          productId,
+          quantity
+        })
+        state.cart.count++
```

Comment:
Co to jest `cart.count`? Czy to jest po prostu liczba elementów w koszyku? Nie lepiej użyć w takim przypadku `cart.products.length`?

Reply:
done.